Source/Blazorise/Styles/_table.scss

```diff
@@ -27,3 +27,68 @@ thead {
         position: relative;
     }
 }
+
+
+/*-- ==============================================================
+ Screen smaller than 760px and iPads.
+ ============================================================== */
+
+@media only screen and (max-width: 760px), (min-device-width: 768px) and (max-device-width: 1024px) {
+
+    [data-content]:before {
+        content: attr(data-content);
+    }
+
+    /* Force table to not be like tables anymore */
+    .b-table-mobile,
+    .b-table-mobile thead,
+    .b-table-mobile tbody,
+    .b-table-mobile th,
+    .b-table-mobile td,
+    .b-table-mobile tr {
+        display: block;
+    }
+
+    .b-table-mobile.text-center {
+        text-align: left !important;
+    }
+
+    .b-table-mobile caption {
+        width: max-content;
+    }
+
+    /* Hide table headers (but not display: none;, for accessibility) */
+    .b-table-mobile thead tr {
+        position: absolute;
+        top: -9999px;
+        left: -9999px;
+    }
+
+    .b-table-mobile > :not(:first-child) {
+        border-top: none;
+    }
+
+    /* Default layout */
+    .b-table-mobile td {
+        /* Behave  like a "row" */
+        border: none;
+        position: relative;
+        padding-top: 1.5rem !important;
+    }
+
+    .b-table-mobile td:before {
```

Comment:
Can we just have a CSS rule for `td` placed inside of `thead`? Would that solve the problem of not need the `data-content`?

Reply:
Can you elaborate?

Reply:
I mean to have `b-table-mobile thead td:before`.

Reply:
The structure is quite different, you have to somehow extract an header value into some other place. Or change the css in a way that somehow each table header cell would be followed by the corresponding table body row cell, which I don't think it's possible with just css.

What I am thinking now, is that doing JS and extracting whatever value is on the header will be more resilient then trying to do it on Blazor/css.

Reply:
> I mean to have `b-table-mobile thead td`.

What does that accomplish differently? Now you would have the rows at head level, and you need to grab the corresponding values from tbody columns somehow.
That's what the data-content trick does basically...

Reply:
I understand the problem now. I agree we have two options.

- Introduce a new API, as you have already done
- Use JS to do the heavylifint

The new API is the best way forward. The user must enable the mobile mode, and adding one more attribute to the table cell should not be much of a problem. Not to mention that it is much easier for us also.

Reply:
Yeah, fot DataGrid it is simple as it will be done automatically. It is only a problem if they construct the table from scratch.

Reply:
Right, but my point was that the DataGrid is going to inherit limitations, also, one of them the Caption Template. :)

Reply:
I forgot the CaptionTemplate 😱

Reply:
Well, should we try the js based approach instead? I'm still conflicted. I feel like the js based approach might allow less limitations regarding the caption at least.